test/components/select/select-screen.test.js

```diff
@@ -337,6 +338,13 @@ describe("Select Screen", () => {
 			eventBus.subscribe.mock.calls[0][0].callback();
 			expect(selectScreen.transientData["test-select"].choice.title).toBe("Title 1");
 		});
+		test("saves choice to local storage", () => {
```

Comment:
might be better name "sets choice to collection" - local storage is kind of abstract here

Reply:
I thought it would be better to be more specific here because you could update 'collection' but not the 'local storage' and viceversa?

Reply:
actually, the point here is to set both :D

Reply:
it's not the concern of the select screen how the data is handled though - that part of the implementation is down to `collection`.  The fact that you are mocking collection highlights this too - if the actual implementation of collection changed then this test would still pass, even if collection started using cookies or a database for example.

Reply:
sorry collection doesn't even care how it's saved either - that's also just a GMI call, so it's only the GMI that handles the localstorage bit.

Obviously we know it's using the local storage, but compartmentalising the units and making everything else a black box is good for unit testing!

Reply:
I've changed it now

Reply:
> the point here is to set both

the method and tests for `setUnique` handles that, so you can assume it's doing that here.

Reply:
sorry for spam, just trying to explain myself best I can 😄